core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -89,12 +89,14 @@ object ZkUtils {
   }
 
   def maybeDeletePath(zkUrl: String, dir: String) {
+    val zk = createZkClient(zkUrl, 30*1000, 30*1000)
     try {
-      val zk = createZkClient(zkUrl, 30*1000, 30*1000)
       zk.deleteRecursive(dir)
-      zk.close()
     } catch {
+      case _: ZkException => zk.deleteRecursive(dir) // Occasionally, child node list is not empty before deleting parent ZkNode, give a second chance to try deleting.
```

Comment:
I'm not sure about this. This is a generic method and I don't think this makes sense generally. I would actually remove `maybeDeletePath` as all callers should really be using `AdminUtils.deleteConsumerGroupInZK`.